src/nvim/eval.c

```diff
@@ -16365,6 +16373,28 @@ void set_vim_var_list(int idx, list_T *val)
     ++val->lv_refcount;
 }
 
+/// Set Dictionary v: variable to "val".
+void set_vim_var_dict(int idx, dict_T *val) FUNC_ATTR_NONNULL_ALL
+{
+  dict_unref(vimvars[idx].vv_dict);
+
+  // Set readonly
+  int todo = (int)val->dv_hashtab.ht_used;
+  for (hashitem_T *hi = val->dv_hashtab.ht_array; todo > 0 ; ++hi) {
+    if (HASHITEM_EMPTY(hi)) {
+       continue;
+    }
+
+    --todo;
+    HI2DI(hi)->di_flags = DI_FLAGS_RO | DI_FLAGS_FIX;
+  }
+
+  vimvars[idx].vv_dict = val;
+  if (val != NULL) {
```

Comment:
A few lines above, `val->dv_hashtab.ht_used` is accessed. If `val` were NULL, it would crash there already. Also, `FUNC_ATTR_NONNULL_ALL` is set. So, `val` cannot be NULL, and checking if it is `NULL` is redundant.